--- ami_cdev.c	2026-05-11 15:52:15.912109982 +0200
+++ /tmp/ami-23.2-patched/ami_cdev.c	2026-05-09 15:01:42.377448690 +0200
@@ -14,6 +14,7 @@
 #include <linux/types.h>
 #include <linux/hwmon.h>
 #include <linux/eventfd.h>
+#include <linux/vmalloc.h>
 
 #include "ami.h"
 #include "ami_hwmon.h"
@@ -41,7 +42,7 @@
  * 
  * Return: NULL.
  */
-static char *devnode(struct device *dev, umode_t *mode)
+static char *devnode(const struct device *dev, umode_t *mode)
 {
 	if (mode)
 		*mode = READ_WRITE;
@@ -721,7 +722,7 @@
 
 	if(!drv_cdev->dev_class) {
 		cls_created = true;
-		drv_cdev->dev_class = class_create(THIS_MODULE, drv_cdev->drv_cls_str);
+		drv_cdev->dev_class = class_create(drv_cdev->drv_cls_str);
 		if (IS_ERR(drv_cdev->dev_class)) {
 			ret = PTR_ERR(drv_cdev->dev_class);
 			PR_ERR("Failed to create class %s. ret : %d",
